src/compiler/checker.ts

```diff
@@ -12520,11 +12520,14 @@ namespace ts {
                     const prop = getPropertyOfType(type, name, skipObjectFunctionPropertyAugment);
                     const modifiers = prop ? getDeclarationModifierFlagsFromSymbol(prop) : 0;
                     if (prop) {
-                        if (isUnion) {
-                            optionalFlag |= (prop.flags & SymbolFlags.Optional);
-                        }
-                        else {
-                            optionalFlag &= prop.flags;
+                        if (prop.flags & SymbolFlags.ClassMember) {

```

Comment:
We should really rename this flag. “Class” threw me off.

```ts
ClassMember = Method | Accessor | Property,
```

Reply:
Yeah, I wanted to rename it, but its not `/* @internal */` so that would be a breaking change. Which I didn't much care to introduce.